mailet/standard/src/main/java/org/apache/james/transport/mailets/MimeDecodingMailet.java

```diff
@@ -52,48 +63,69 @@
 
     public static final String ATTRIBUTE_PARAMETER_NAME = "attribute";
 
-    private String attribute;
+    private AttributeName attribute;
 
     @Override
     public void init() throws MessagingException {
-        attribute = getInitParameter(ATTRIBUTE_PARAMETER_NAME);
-        if (Strings.isNullOrEmpty(attribute)) {
+        String attributeRaw = getInitParameter(ATTRIBUTE_PARAMETER_NAME);
+        if (Strings.isNullOrEmpty(attributeRaw)) {
             throw new MailetException("No value for " + ATTRIBUTE_PARAMETER_NAME
                     + " parameter was provided.");
         }
+        attribute = AttributeName.of(attributeRaw);
     }
 
     @Override
     public void service(Mail mail) throws MessagingException {
-        if (mail.getAttribute(attribute) == null) {
+        if (!mail.getAttribute(attribute).isPresent()) {
             return;
         }
 
-        ImmutableMap.Builder<String, byte[]> extractedMimeContentByName = ImmutableMap.builder();
-        for (Map.Entry<String, byte[]> entry: getAttributeContent(mail).entrySet()) {
-            Optional<byte[]> maybeContent = extractContent(entry.getValue());
-            if (maybeContent.isPresent()) {
-                extractedMimeContentByName.put(entry.getKey(), maybeContent.get());
-            }
-        }
-        mail.setAttribute(attribute, extractedMimeContentByName.build());
+        mail.setAttribute(new Attribute(attribute, AttributeValue.of(getExtractedContent(mail))));
+    }
+
+    private ImmutableMap<String, AttributeValue<?>> getExtractedContent(Mail mail) throws MailetException {
+        return AttributeUtils
+                .getValueAndCastFromMail(mail, attribute, Serializable.class)
+                .map(this::extractAttributeContent)
+                .orElse(ImmutableMap.<String, AttributeValue<?>>of());
     }
 
     @SuppressWarnings("unchecked")
-    private Map<String, byte[]> getAttributeContent(Mail mail) throws MailetException {
-        Serializable attributeContent = mail.getAttribute(attribute);
+    private ImmutableMap<String, AttributeValue<?>> extractAttributeContent(Serializable attributeContent) {
         if (! (attributeContent instanceof Map)) {
             LOGGER.debug("Invalid attribute found into attribute {} class Map expected but {} found.",
                     attribute, attributeContent.getClass());
             return ImmutableMap.of();
         }
-        return (Map<String, byte[]>) attributeContent;
+
+        Map<String, AttributeValue<?>> attributeMap = (Map<String, AttributeValue<?>>) attributeContent;
+        ThrowingFunction<Entry<String, AttributeValue<?>>, Optional<Pair<String, AttributeValue<BytesArrayDto>>>> performExtraction =
```

Comment:
this statement is really hard to understand IMO, only a type declaration takes a whole line. Consider my proposal:
```java
    private ImmutableMap<String, AttributeValue<?>> extractAttributeContent(Serializable attributeContent) {
        ...
        Map<String, AttributeValue<?>> attributeMap = (Map<String, AttributeValue<?>>) attributeContent;

        return attributeMap
            .entrySet()
            .stream()
            .filter(this::convertAbleEntry)
            .map(Throwing.function(this::convertToByteArrayDto).sneakyThrow())
            .filter(Optional::isPresent)
            .map(Optional::get)
            .collect(ImmutableMap.toImmutableMap(Pair::getKey, pair -> AttributeValue.of(pair.getValue())));
    }

    private boolean convertAbleEntry(Entry<String, AttributeValue<?>> entry) {
        Object value = entry.getValue().getValue();
        return value instanceof BytesArrayDto || value instanceof String;
    }

    private Optional<Pair<String, BytesArrayDto>> convertToByteArrayDto(Entry<String, AttributeValue<?>> entry) throws MessagingException {
        byte[] bytes = convertToByteArray(entry.getValue().getValue());
        return extractContent(bytes)
            .map(bytesArrayDto -> Pair.of(entry.getKey(), bytesArrayDto));
    }

    private byte[] convertToByteArray(Object value) {
        if (value instanceof BytesArrayDto) {
            return ((BytesArrayDto) value).getValues();
        } else {
            return ((String) value).getBytes(StandardCharsets.UTF_8);
        }
    }
```